### ISSUE TITLE
encoding jpeg2000 errors out

### ISSUE THREAD

**Issue by bradh** [strukturag/libheif#1309]
While testing for a different problem related to JPEG 2000 decoding, I hit this:

```
./examples/heif-enc --jpeg2000 ~/testbed20/silvereye/smalltiles_2024-06-29/simple_osm_tile.png --output testfile.heij
heif-enc: /home/bradh/coding/libheif/libheif/box.cc:3119: void Box_clap::set(uint32_t, uint32_t, uint32_t, uint32_t): Assertion `image_width >= clap_width' failed.
Aborted (core dumped)
```

It doesn't seem input file specific, but will attach.

```
$ ./examples/heif-enc --list-encoders
AVC encoders:
AVIF encoders:
- aom = AOMedia Project AV1 Encoder v3.8.2 [default]
HEIC encoders:
- x265 = x265 HEVC encoder (3.5+1-f0c1022b6) [default]
JPEG encoders:
JPEG 2000 encoders:
- openjpeg = OpenJPEG 2.5.0 [default]
JPEG 2000 (HT) encoders:
Uncompressed: no
VVIC encoders:
```

**Comment by bradh** [comment]
Test file

![simple_osm_tile](https://github.com/user-attachments/assets/d261a8c7-22e6-441b-a623-a4e0fbdafb69)

**Comment by farindk** [comment]
Thanks. Fixed this in the plugin and also fixed parsing the J2K header (in a680e3d).
